### ISSUE TITLE
Naming

### ISSUE THREAD

**Issue by Kestrer** [programatik29/axum-server#1]
axum-server isn't an ideal name because this project is not strictly related to Axum, and that can be confusing. This issue is for discussion regarding options for different names.

Some initial ideas:
- tower-hyper-server: Do we want to tie this library inherently to Hyper? This name is also long.
- tower-http-server: Another long name.

**Comment by david-perez** [comment]
I agree with @Kestrer that the project's name is not ideal since there really isn't anything in it that intrinsically ties it to axum. I think the project's adoption would grow if the useful abstractions that it exposes, especially around TLS management, were upstreamed into separate crates or somewhere in the Hyper/Tower/rustls projects.

**Comment by andrewbanchich** [comment]
I ran into the same confusion [here](https://github.com/tokio-rs/axum/issues/677#issuecomment-1003374883).

I also agree with @david-perez and think, even from a security standpoint, people would be more comfortable adopting this if it were under the Hyper / Tokio / Rustls umbrella.

**Comment by programatik29** [comment]
I would like to wait until [this issue](https://github.com/hyperium/hyper/issues/2321) is resolved before renaming.

**Comment by programatik29** [comment]
I think `hyperion` would be a good new name for this crate. What do you think?

**Comment by jplatte** [comment]
Very similar to "hyperium", the org under which hyper and a few other crates are hosted on GitHub. But otherwise I like it.

**Comment by programatik29** [comment]
This library will always be high level server **i**mplementation **on** **hyper** thus **hyperion**. In my opinion it shouldn't be confusing for rust community.

**Comment by josb** [comment]
**hyperion** is at least a less confusing name than axum-server, given that this project is not strictly related to axum.

**Comment by programatik29** [comment]
Depending on the (future) scope of `hyper-util`, it might not be necessary to continue this project if `hyper-util` covers all use cases. There might still be a gap for complete server implementation (TLS, utilities and higher level implementations) after `hyper-util` is complete then I will rename this crate to `hyperion` and continue development.
